### PR TITLE
Trivial cleanups

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Review by bluca — APPROVED** [review]
LGTM - will remove the sysext change from my other PR

**Comment by keszybz** [comment]
> Building SRPM ...

Oh, packit attaches comments to whole commits. This was possible in the past in the web ui, but went away a few years ago. It's annoying not to be able to comment on the commit message.

> LGTM - will remove the sysext change from my other PR

I see now that you made the result ignored. I didn't do that. Maybe we still want to this... We'll figure it out ;)

**Comment by mrc0mmand** [comment]
> > Building SRPM ...
> 
> Oh, packit attaches comments to whole commits. This was possible in the past in the web ui, but went away a few years ago. It's annoying not to be able to comment on the commit message.

That's actually a bug, where Packit doesn't cancel the previous builds, so sometimes when they finish after a force push, they'll comment the result to the old commit instead of sending the results to /dev/null (see https://github.com/packit/packit-service/issues/965).

File: src/shared/extension-release.c [hunk 1]
```diff
@@ -79,16 +80,46 @@ int extension_release_validate(
...
-        r = getenv_path_list("SYSTEMD_SYSEXT_HIERARCHIES", ret_hierarchies);
-        if (r < 0)
-                return log_debug_errno(r, "Failed to parse SYSTEMD_SYSEXT_HIERARCHIES environment variable : %m");
-        if (!*ret_hierarchies) {
-                *ret_hierarchies = strv_new("/usr", "/opt");
-                if (!*ret_hierarchies)
-                        return -ENOMEM;
+        assert(ret_hierarchies);
+
+        e = secure_getenv("");
```
Comment (poettering) [inline src/shared/extension-release.c]: hmm, wah?

File: src/basic/env-util.c [hunk 2]
```diff
@@ -803,44 +802,3 @@ int setenv_systemd_exec_pid(bool update_only) {
 
         return 1;
 }
-
-int getenv_path_list(const char *name, char ***ret_paths) {
```
Comment (poettering) [inline src/basic/env-util.c]: i think this should probably stay generic and we should use it in xdg_user_dirs() and such things. There's plenty places we parse env vars that are lists of paths, and hence we should have a generic implementation i am sure

File: src/resolve/resolved-dns-stub.c [hunk 3]
```diff
@@ -89,15 +89,6 @@ DnsStubListenerExtra *dns_stub_listener_extra_free(DnsStubListenerExtra *p) {
...
 }
 
-uint16_t dns_stub_listener_extra_port(DnsStubListenerExtra *p) {
-        assert(p);
-
-        if (p->port > 0)
-                return p->port;
-
-        return 53;
-}
```
Comment (poettering) [inline src/resolve/resolved-dns-stub.c:99]: i wouldn't really bother with this too much. given lto the compiler should figure this out on its own, whether its worth inlining this

**Review by poettering — CHANGES_REQUESTED** [review]

**Comment by poettering** [comment]
also see #18798

File: src/shared/extension-release.c (hunk 1 above)
Comment (keszybz) [reply src/shared/extension-release.c]: What? Somebody's doing crazy stuff!

File: src/resolve/resolved-dns-stub.c (hunk 3 above)
Comment (keszybz) [reply src/resolve/resolved-dns-stub.c:99]: It makes the code shorter and thus easier to follow.

**Comment by keszybz** [comment]
OK, I dropped the first commit. I'll resubmit it if appropriate after #18798 is merged.
Restoring the green label.

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**